lib/util/createDomain.js

```diff
@@ -1,13 +1,14 @@
 "use strict";
 const url = require("url");
+const getLocalIP = require("./getLocalIP");
 
 module.exports = function createDomain(options) {
 	const protocol = options.https ? "https" : "http";
 
 	// the formatted domain (url without path) of the webpack server
 	return options.public ? `${protocol}://${options.public}` : url.format({
 		protocol: protocol,
-		hostname: options.host,
+		hostname: options.lan ? getLocalIP() : options.host,
```

Comment:
I don't feel that `lan` is semantic for this feature. `options.useLocalIp` or some such would probably be easier to understand at first glance.